posthog/async_migrations/migrations/0004_replicated_schema.py

```diff
@@ -173,22 +176,29 @@ def replicated_table_operations(self, table: TableMigrationData):
         )
 
     def finalize_table_operations(self, table: TableMigrationData):
+
         # NOTE: Relies on IF NOT EXISTS on the query
         if isinstance(table, ShardedTableMigrationData):
             for table_name, create_table_query in table.extra_tables.items():
                 yield AsyncMigrationOperationSQL(sql=create_table_query, rollback=f"DROP TABLE IF EXISTS {table_name}")
 
-        if isinstance(table, ShardedTableMigrationData) and table.materialized_view_name is not None:
-            yield AsyncMigrationOperationSQL(sql=f"DROP TABLE IF EXISTS {table.materialized_view_name}", rollback=None)
+            for mv_name in table.materialized_view_names:
+                yield AsyncMigrationOperationSQL(sql=f"DROP TABLE IF EXISTS {mv_name}", rollback=None)
 
-        if table.kafka_table_name is not None:
+        for i in range(len(table.kafka_table_names)):
+            create_kafka_table = table.create_kafka_tables[i]
+            kafka_table_name = table.kafka_table_names[i]
             yield AsyncMigrationOperationSQL(
-                sql=cast(str, table.create_kafka_table), rollback=f"DROP TABLE IF EXISTS {table.kafka_table_name}"
+                sql=cast(str, create_kafka_table), rollback=f"DROP TABLE IF EXISTS {kafka_table_name}"
             )
 
-        if isinstance(table, ShardedTableMigrationData) and table.materialized_view_name is not None:
+        if isinstance(table, ShardedTableMigrationData):
+            for i in range(len(table.materialized_view_names)):
+                create_mv = table.create_materialized_views[i]
+                mv_name = table.materialized_view_names[i]
+
             yield AsyncMigrationOperationSQL(
```

Comment:
Bug: This has the wrong indentation. Only the last mat view is dropped.

Reply:
good catch